src/compiler/scala/tools/nsc/transform/Delambdafy.scala

```diff
@@ -294,12 +300,25 @@ abstract class Delambdafy extends Transform with TypingTransformers with ast.Tre
       case Template(_, _, _) =>
         def pretransform(tree: Tree): Tree = tree match {
           case dd: DefDef if dd.symbol.isDelambdafyTarget =>
-            if (!dd.symbol.hasFlag(STATIC) && methodReferencesThis(dd.symbol)) {
+            val ddef = if (!dd.symbol.hasFlag(STATIC) && methodReferencesThis(dd.symbol)) {
               gen.mkStatic(dd, dd.symbol.name, sym => sym)
             } else {
               dd.symbol.setFlag(STATIC)
               dd
             }
+            if (settings.Ydelambdafy.value == "method-ref")
+              ddef match {
+                case DefDef(_, _, _, List(self :: Nil), _, Apply(select @ Select(core: Ident, _), Nil))
```

Comment:
Would be good to document these conditions and suggest any potential ways to relax them.

Reply:
Document where?

And I'm not sure what should be possible and I'm just not considering, and what would be nice, but would be too much to chew right now.  Test cases ideas are very welcome!  (I've already minimised and incorporated a few by running this unconditionally on the whole test suite.)

Reply:
Yep, a comment right here that explains the assumptions behind the conditions. Expected tree shape is also an excellent suggestion for docs.